elastic/kibana
# [Remote Clusters] Improve Connection mode form fields

Closes https://github.com/elastic/kibana/issues/162500

## Summary

This PR improves the Connection mode fields of the Remote cluster creation form on Cloud.

<img width="1245" alt="Screenshot 2024-02-12 at 11 10 27" src="https://github.com/elastic/kibana/assets/59341489/7cdec999-bf74-406c-8f67-fa5f7bdcd424">

When we switch the toggle on:
<img width="1245" alt="Screenshot 2024-02-12 at 11 13 01" src="https://github.com/elastic/kibana/assets/59341489/f849883f-e876-4fdf-9272-af4c83e4ee24">

**How to test:**
These changes apply only for on-Cloud so here are instructions for testing them in a cloud deployment:
1. If you haven't done yet, configure Vault locally (https://docs.elastic.dev/ci/using-secrets#installing-vault).
2. Run `export VAULT_ADDR=https://secrets.elastic.co:8200`
3. Log into Vault: `vault login --method github`
4. Obtain the cloud deployment credentials: `vault read secret/kibana-issues/dev/cloud-deploy/kibana-pr-176547`
5. Access the CI Cloud deployment for this PR: https://kibana-pr-176547.kb.us-west2.gcp.elastic-cloud.com:9243/
6. Log in with the obtained credentials.
7. Go to Stack Management -> Remote Clusters and start creating a new remote cluster.
8. Try different inputs in the Connection mode fields, click on "Show Request" and verify that the request is correct.
9. Verify that not adding a port to the remote address will result in the `9400` port in the request.
10. Verify that if the `https://` protocol is added to the remote address, it will be stripped in the request.
11. Verify that the Server name by default is equal to the host from the remote address unless you specify a different server name.
13. Verify that a missing remote address or one that has host with invalid characters results in an error message.
14. Start editing a created remote cluster and verify that the form is correctly filled.

The self-managed version should not be affected by these changes. Run Es locally with `yarn es snapshot` and Kibana with `yarn start` and verify that the Connection mode fields look and behave in the same way as they do now.
